### PR TITLE
[RFC] Draw annotations

### PR DESCRIPTION
**Pull request by Unia**
1. Replace the black rectangle with an overview of the genes.
2. Upon zooming in, move genes down and show where on the strain they are connected.

Issues:
1. Connections between strain and annotations move when panning
2. Drawing annotations on the cluster-level is **ugly**, inefficient and chaotic. We have to come up with a better way to implement this next sprint.

### CODE REVIEW

File: dnainator-javafx/src/main/java/nl/tudelft/dnainator/javafx/drawables/SemanticDrawable.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+		this.childContent = childContent;
+		getChildren().addAll(content, childContent);
+	}
+
+	/**
+	 * Load the {@link Drawable} content of the view itself. Must be called by the constructor of
+	 * the overriding class!
+	 * @param bounds The bounds of the viewport.
+	 */
+	public abstract void loadContent(Bounds bounds);
```
Comment (pvanagtmaal) [inline dnainator-javafx/src/main/java/nl/tudelft/dnainator/javafx/drawables/SemanticDrawable.java]: probably better to be protected, same for the other methods in this class.

File: dnainator-javafx/src/main/java/nl/tudelft/dnainator/javafx/drawables/strains/Strain.java [hunk 2]
```diff
@@ -51,60 +53,109 @@ public Strain(ColorServer colorServer, Graph graph) {
...
-			content.getChildren().add(rectangle);
-		}
+	@Override
+	public void loadContent(Bounds bounds) {
+		content.getChildren().clear();
+		HBox hbox = new HBox();
+		Range ranks = getRange(bounds);
+
+		List<Annotation> annotations = graph.getAnnotationByRank(ranks).stream()
+				.sorted((a1, a2) -> (a1.getStart() < a2.getStart()) ? -1 : 1)
```
Comment (pvanagtmaal) [inline dnainator-javafx/src/main/java/nl/tudelft/dnainator/javafx/drawables/strains/Strain.java]: won't pass check style.

**Comment by pvanagtmaal** [comment]
Good for now, working well enough to merge.

**Comment by Balletie** [comment]
Yes, this is was a big task, so it's best to merge this for now and continue working on the task next week. Merging.

**Comment by ohuang12** [comment]
Fine to me!

**Pull request merged**